source/geod24/LocalRest.d

```diff
@@ -351,180 +334,449 @@ public final class RemoteAPI (API) : API
+            });
+
+        }
+
+        return res;
     }
 
-    /// Where to send message to
-    private C.Tid childTid;
+}
+
+/*******************************************************************************
+
+    Provide eventloop-like functionalities
+
+    Since nodes instantiated via this modules are Vibe.d server,
+    they expect the ability to run an asynchronous task ,
+    usually provided by `vibe.core.core : runTask`.
+
+    In order for them to properly work, we need to integrate them to our event
+    loop by providing the ability to spawn a task, and wait on some condition,
+    optionally with a timeout.
+
+    The following functions do that.
+    Note that those facilities are not available from the main thread,
+    while is supposed to do tests and doesn't have a scheduler.
+
+*******************************************************************************/
+
+public void runTask (void delegate() dg)
+{
+    assert(thisScheduler !is null, "Cannot call this function from the main thread");
+    thisScheduler.spawn(dg);
+}
+
+
+/// Ditto
+public void sleep (Duration timeout)
+{
+    assert(thisScheduler !is null, "Cannot call this function from the main thread");
+    scope c = thisScheduler.newCondition(null);
+    thisScheduler.wait(c, timeout);
+}
+
 
-    /// Whether or not the destructor should destroy the thread
-    private bool owner;
+/*******************************************************************************
+
+    It has one `Server` and one `Client`. make a new thread It uses `Server`.
+
+*******************************************************************************/
+
+public class RemoteAPI (API) : API
+{
+    /***************************************************************************
+
+        Instantiate a node and start it
+
+        This is usually called from the main thread, which will start all the
+        nodes and then start to process request.
+        In order to have a connected network, no nodes in any thread should have
+        a different reference to the same node.
+        In practice, this means there should only be one `Transceiver`
+        per "address".
+
+        Note:
+          When the `RemoteAPI` returned by this function is finalized,
+          the child thread will be shut down.
+          This ownership mechanism should be replaced with reference counting
+          in a later version.
+
+        Params:
+          Impl = Type of the implementation to instantiate
+          args = Arguments to the object's constructor
+          timeout = (optional) timeout to use with requests
+
+        Returns:
+          A `RemoteAPI` owning the node reference
+
+    ***************************************************************************/
+
+    public static RemoteAPI!(API) spawn (Impl) (CtorParams!Impl args,
+        Duration timeout = Duration.init)
+    {
+        auto server = Server!API.spawn!Impl(args);
+        return new RemoteAPI(server.transceiver, timeout);
```

Comment:
Why is there a `Server` and a `RemoteAPI` ? What's the difference ? They seem to be doing essentially the same things, but differently. For example `Server` does not handle timeout.